cognite/neat/_issues/_base.py

```diff
@@ -215,207 +208,34 @@ def __eq__(self, other: object) -> bool:
 class NeatError(NeatIssue, Exception):
     """This is the base class for all exceptions (errors) used in Neat."""
 
-    @classmethod
-    def from_errors(cls, errors: "list[ErrorDetails | NeatError]", **kwargs) -> "list[NeatError]":
-        """Convert a list of pydantic errors to a list of Error instances.
-
-        This is intended to be overridden in subclasses to handle specific error types.
-        """
-        all_errors: list[NeatError] = []
-        read_info_by_sheet = kwargs.get("read_info_by_sheet")
-
```

Comment:
All factory methods moved to its own module to avoid circular dependencies and adhere to single responsibility. This module now only have base classes, while factory module is for creating issues.